oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -137,6 +139,50 @@ protected ExternalAccountCredentials(
     this.clientSecret = clientSecret;
     this.scopes =
         (scopes == null || scopes.isEmpty()) ? Arrays.asList(CLOUD_PLATFORM_SCOPE) : scopes;
+    this.environmentProvider = SystemEnvironmentProvider.getInstance();
```

Comment:
Let's convert this constructor to calling the one with `EnvironmentProvider`. We usually don't want 2 constructors doing logic if possible.